packages/notifi-react-card/lib/components/intercom/ChatMessageSection.tsx

```diff
@@ -1,40 +1,96 @@
 import React from 'react';
+import { Virtuoso } from 'react-virtuoso';
 
-import { ChatMessageDate, ChatMessageDateProps } from './ChatMessageData';
+import { useIntercomChat } from '../../hooks/useIntercomChat';
+import { ChatMessageDate, ChatMessageDateProps } from './ChatMessageDate';
+import {
+  ChatWindowIntroSection,
+  ChatWindowIntroSectionProps,
+} from './ChatWindowIntroSection';
 import { MessageList, MessageListProps } from './MessageList';
 
 export type ChatMessageSectionProps = Readonly<{
   classNames?: Readonly<{
-    container: string;
-    content: string;
-    date: ChatMessageDateProps['classNames'];
+    chatWindowIntro: ChatWindowIntroSectionProps['classNames'];
     messageList: MessageListProps['classNames'];
+    date: ChatMessageDateProps['classNames'];
   }>;
   chatMessageSectionIntroContent?: string;
 }>;
 
 export const ChatMessageSection: React.FC<ChatMessageSectionProps> = ({
   classNames,
-  chatMessageSectionIntroContent = 'What can we help you with today?',
+  chatMessageSectionIntroContent,
 }) => {
+  const {
+    conversation,
+    setIsScrolling,
+    setVisibleRange,
+    setAtTop,
+    isLoading,
+    hasNextPage,
+  } = useIntercomChat({
+    conversationId: '8c5ccf41d93c4bb9b5e1e1014002e923',
+  });
+
   return (
-    <div
-      className={clsx(
-        'NotifiIntercomChatMessageSection__container',
-        classNames?.container,
+    <>
+      {conversation.feed.length === 0 ? (
+        <ChatWindowIntroSection
+          classNames={classNames?.chatWindowIntro}
+          startDate={conversation.createdDate}
+          chatMessageSectionIntroContent={chatMessageSectionIntroContent}
+        />
+      ) : null}
+
+      {conversation.feed.length === 0 ? null : (
+        <Virtuoso
+          atTopStateChange={setAtTop}
+          className="virtual-container"
+          data={conversation.feed}
+          followOutput="auto"
+          isScrolling={setIsScrolling}
+          rangeChanged={setVisibleRange}
+          itemContent={(index, feed) => {
+            return (
+              <div key={index}>
+                {isLoading ? (
+                  <div
+                    style={{
+                      display: 'flex',
+                      justifyContent: 'center',
+                      padding: '10px',
+                    }}
+                  >
+                    Loading...
+                  </div>
+                ) : null}
+                {index === 0 && !hasNextPage && (
+                  <ChatWindowIntroSection
+                    classNames={classNames?.chatWindowIntro}
+                    startDate={conversation.createdDate}
+                    chatMessageSectionIntroContent={
+                      chatMessageSectionIntroContent
+                    }
+                    inVirtualContainerStyle={
+                      'ChatWindowIntro__virtualContainer'
+                    }
+                  />
+                )}
+                {index != 0 && (
```

Comment:
How are you showing the date?

Though it is not visible in the demo, I wonder if this may cause the date to render more than needed. I interpret this as if the index is anything other than at the state, render a date.

Reply:
the date has been rendered for each feed, we don't need the date to be rendered when index is equal to 0, cause we have the created conversation date in the intro section.